tests/main/interfaces-gpg-keys/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure that the gpg-keys interface works.
+
+details: |
+    The gpg-keys interface allows to access gpg binary and keys.
+
+environment:
+    KEYSDIR: "$HOME/.gnupg"
+    TRUSTDB: "$HOME/.gnupg/trustdb.gpg"
+    TESTKEY: "$HOME/.gnupg/testkey"
+    CONNECTED_PATTERN: ':gpg-keys +test-snapd-gpg-keys'
+    DISCONNECTED_PATTERN: '\- +test-snapd-gpg-keys:gpg-keys'
+
+prepare: |
+    . $TESTSLIB/snaps.sh
+    install_local test-snapd-gpg-keys
+
+    if [ -d "$KEYSDIR" ]; then
+        cp -rf "$KEYSDIR" "$KEYSDIR".old
+    else
+        mkdir "$KEYSDIR"
+    fi
+
+    if ! [ -f $TRUSTDB ]; then
+        echo "trustdb" > "$TRUSTDB"
+    fi
+
+    echo "testkey" > "$TESTKEY"
+
+restore: |
+    rm -f call.error
+    rm -rf "$KEYSDIR"
+    if [ -d "$KEYSDIR".old ]; then
+        mv "$KEYSDIR".old "$KEYSDIR"
+    fi
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "When the interface is connected"
+    snap connect test-snapd-gpg-keys:gpg-keys
+
+    echo "Then the snap is able to run gpg"
+    test-snapd-gpg-keys.sh -c "gpg --list-keys"
```

Comment:
Hmm, did you forget to add `test-snapd-gpg-keys.sh` file?

Reply:
No, the `.sh` file is just shell itself.

Reply:
Ah, of course! I got confused, thanks!